### PR TITLE
Quick Actions: Implement shortcut for adding new text element

### LINKED ISSUE #10974: UI Simplification: Insert a text when using the `T+` quick-action
## Feature Description
Previously, we had a `T+` icon inside the Insertion panel Text tab to add a text quickly. This was removed in #10493.
UX wants to bring back an option to insert a text quickly, but this will now be done from the quick-action menu.

Todo:

- [ ] Adjust the `T+` button in the quick-action menu so that it would insert a new text **in addition** to its current functionality (opening the Style panel and highlighting the presets area)
- [ ] The text content will be `Fill in some text` and it will have the same props as the [previous quick-action text had.](https://github.com/GoogleForCreators/web-stories-wp/pull/10493/files#diff-2ce9cbec9cb7bf2cc94d4207e01993a95353ae7cee46efe191f5ec86e830b8f8R57) We'll need to add the preset back to the text presets file.
- [ ] Add Karma tests for the new feature.

Note that everything else stays the same: when to display the button, its previous functionality will remain, etc.

## Additional Context

Props the text was previously added with:
```
const DEFAULT_PRESET = {
  content: __('Fill in some text', 'web-stories'),
  fontWeight: FONT_WEIGHT.NORMAL,
  fontSize: dataFontEm(1.2),
  lineHeight: 1.5,
  x: DEFAULT_LEFT_MARGIN,
  y: (PAGE_HEIGHT - dataFontEm(1.5)) / 2,
  font: TEXT_ELEMENT_DEFAULT_FONT,
  width: DEFAULT_WIDTH.DEFAULT,
  textAlign: 'center',
  borderRadius: DEFAULT_TEXT_BORDER_RADIUS,
  padding: DEFAULT_TEXT_PADDING,
};
```

T+ button in the quick-action menu:
Screenshot 2022-03-17 at 16 53 50

### PR DESCRIPTION
**Pull request by marianakyrkosh**
## Context

 T+ icon inside the Insertion panel Text tab should add a text quickly. 

## Summary

Clicking the T+ icon adds a paragraph to the canvas  and opens the Text panel on the left (as it was before).
It also adds a trackEvent 'library_text_quick_action'.

## User-facing changes

image
image

## Testing Instructions

This PR can be tested by following these steps:

1. Click on the T+ button.
2. Text appears on the canvas.
3. Text panel opens on the left.
4. A trackEvent 'library_text_quick_action' is added.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10974

### CODE REVIEW

File: packages/story-editor/src/components/floatingMenu/elements/karma/text.karma.js [hunk 1]
```diff
@@ -193,7 +191,7 @@ describe('Design Menu: Text Styles', () => {
       await fixture.events.keyboard.press('tab');
 
       // Select character 7 and 8 (the part "ip" in "Lorem ipsum dolor sit amet, consectetur adipiscing elit.")
-      await setSelection(6, 8);
+      await setSelection(5, 7);
```
Comment (merapi) [inline packages/story-editor/src/components/floatingMenu/elements/karma/text.karma.js]: Nit: change the comment to new text (here, and I saw it also elsewhere).

**Review by merapi — COMMENTED** [review]
Looks good overall but it should probably be `Fill in some text` (without dot) at least that's how it's mentioned in the ticket/on Slack/in the old version.

File: packages/story-editor/src/components/canvas/pageSideMenu.js [hunk 2]
```diff
@@ -71,6 +74,12 @@ function PageSideMenu() {
   const quickActions = useQuickActions();
 
   const isZoomed = zoomSetting !== ZOOM_SETTING.FIT;
+  const insertElement = useInsertElement();
+  const handleAddText = (evt) => {
+    evt.stopPropagation();
+    insertElement('text', DEFAULT_PRESET);
+    trackEvent('library_text_quick_action');
```
Comment (swissspidy) [inline packages/story-editor/src/components/canvas/pageSideMenu.js]: This should follow the naming convention for other Quick Actions tracking events.

e.g.

```js
trackEvent('quick_action', {
  name: ACTIONS.INSERT_TEXT_QUICK_ACTION.trackingEventName,
}
```

File: tests/phpunit/integration/data/story_post_content_filtered.json [hunk 3]
```diff
@@ -77,7 +77,7 @@
           },
           "type": "text",
           "id": "e781bc30-4ebe-416f-bf1d-b509dc3ef6ce",
-          "content": "<span style=\"font-weight: 700\">Fill in some text</span>",
+          "content": "<span style=\"font-weight: 700\">.</span>",
```
Comment (swissspidy) [inline tests/phpunit/integration/data/story_post_content_filtered.json]: Is this intentional?

File: packages/story-editor/src/components/canvas/pageSideMenu.js (hunk 2 above)
Comment (marianakyrkosh) [reply packages/story-editor/src/components/canvas/pageSideMenu.js]: Done

File: tests/phpunit/integration/data/story_post_content_filtered.json (hunk 3 above)
Comment (marianakyrkosh) [reply tests/phpunit/integration/data/story_post_content_filtered.json]: Done

File: packages/story-editor/src/components/floatingMenu/elements/karma/text.karma.js (hunk 1 above)
Comment (marianakyrkosh) [reply packages/story-editor/src/components/floatingMenu/elements/karma/text.karma.js]: Done

**Comment by mariana-k** [comment]
> Looks good overall but it should probably be `Fill in some text` (without dot) at least that's how it's mentioned in the ticket/on Slack/in the old version.

Removed the dot

**Comment by merapi** [comment]
I've overlooked one thing: the insert logic should be in [useQuickActions.js](https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/story-editor/src/app/highlights/quickActions/useQuickActions.js#L464), not in `pageSideMenu.js` (it's a generic component) and since we already have `trackEvent` there - the new one is not needed? @swissspidy

**Comment by spacedmonkey** [comment]
There seems to be valid test failures here. Can we look into these. Otherwise, this change seems good to me.

**Comment by mariana-k** [comment]
> I've overlooked one thing: the insert logic should be in [useQuickActions.js](https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/story-editor/src/app/highlights/quickActions/useQuickActions.js#L464), not in `pageSideMenu.js` (it's a generic component) and since we already have `trackEvent` there - the new one is not needed? @swissspidy

done

File: packages/story-editor/src/components/canvas/pageSideMenu.js [hunk 4]
```diff
@@ -116,6 +116,7 @@ function PageSideMenu() {
                         aria-label={label}
                         onClick={(evt) => {
                           onClick(evt);
+
```
Comment (merapi) [inline packages/story-editor/src/components/canvas/pageSideMenu.js:124]:
```suggestion
```

**Review by merapi — APPROVED** [review]
✅

**Comment by swissspidy** [comment]
FYI some JS unit tests are failing now

**Comment by mariana-k** [comment]
> FYI some JS unit tests are failing now

Fixed JS unit tests

**Pull request merged**